public/javascript/main.js

```diff
@@ -478,9 +492,61 @@ $(function(){
     }
 
   }
+
+  if (typeof pageInitFunction === 'function') {
+    pageInitFunction();
+  }
 });
 
 function getQueryParam(key) {
   var match = location.search.match(new RegExp("[?&]" + key + "=([^&]+)(&|$)"));
   return match && decodeURIComponent(match[1].replace(/\+/g, " "));
 }
+
+// collapseDisplayedItems:
+// If there are too many items being displayed in the list, hide
+// all but a few, making sure the target element (if there is one)
+// is included in those that are not hidden. Add a button for
+// revealing the hidden items.
+// Note this is used on the download page, where (currently) there
+// may be more than one such list (e.g., one for each legislature).
+
+function collapseDisplayedItems(
+    $ul,               // unordered list elementthat needs to be collapsed
+                       // note: <ul> must have an id
+    $targetItem,       // item within that which needs to be highlighted
+                       // ...if none (which is OK) take the first
+    hiddenClassName,   // CSS class used to distinguish collapsed elements
+    minThresholdItems, // only collapse if more elements than this
+    maxDisplayItems,   // number of elements to show when collapsed
+    buttonText         // message on reveal button
+  ) {
+  var $listItems = $ul.find('li');
+  if ($listItems.length > minThresholdItems) {
+    $listItems.addClass(hiddenClassName);
+    // want a slice around targetIndex of maxDisplayItems
+    var targetIndex = $listItems.index($targetItem);
+    var maxUpperBound = $listItems.length;
```

Comment:
I'm finding this logic a little too difficult to follow, and it interrupts the flow of this function quite severely. This seems like it should really be a separate function that either partitions a list for you, or even perhaps just one that gives you the indices you want. (It also seems much more complex that I would expect. Is that simply "because Javascript"?)